index.html

```diff
@@ -0,0 +1,464 @@
+              not navigated to the launch's target URL, instead the target URL
+              is communicated via {{LaunchParams}} . If there are no existing
+              web app clients the [=route to/new-client=] behaviour is used
+              instead.
+            </dd>
+          </dl>
+        </p>
+        <p>
+          To <dfn>process the `route_to` member</dfn>, given [=ordered
+          map=] |json_launch_handler:ordered map|, [=ordered map=]
+          |manifest_launch_handler:ordered map|, run the following:
+        </p>
+        <ol class="algorithm">
+          <li>If |json_launch_handler|["route_to"] does not [=map/exist=],
+          return.
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is
+          [=list=]:
+            <ol class="algorithm">
+              <li>[=list/For each=] |entry| of |json_launch_handler|["route_to"]:
+                <ol class="algorithm">
+                  <li>If the type of |entry| is not [=string=], continue.
+                  </li>
+                  <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |entry| and return.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is not [=string=], return.
+          </li>
+          <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |json_launch_handler|["route_to"].
+          </li>
+        </ol>
+        <p class="note">
+          `route_to` accepts a list of strings as a future proofing mechanism. If a new [=route to target=] is added in the future but has not yet been implemented by all user agents, web developers can cater to the old user agents by providing fallback [=route to target=] values in a list format. User agents will use the first value that is valid to them.
+        </p>
+        <p class="note">
+          For example: if "some-new-behaviour" is added in the future a site developer can specify "[=route to/existing-client-navigate=]" to be the fallback with the following:
+        </p>
+        <pre class="note json">
+          {
+            "launch_handler": {
+              "route_to": ["some-new-behaviour", "existing-client-navigate"]
+            }
+          }
+        </pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Handling Web App Launches
+      </h2>
+      <section>
+        <h2>
+          Launching a Web App
+        </h2>
+        <p>
+          When launching a web app via any means (e.g. OS launch, protocol handling, file handling, share target, link capturing, etc.) the trigger of the launch should run the steps to [=launch a web app=] with a {{LaunchParams}} containing relevant launch information (e.g. share data in the event of a share target launch).
```

Comment:
I think this "eg" parenthetical is too vague (i.e. not linked to appropriate specs which may or may not yet exist, and is non-exhaustive) to be appropriate in normative text. I would remove the parenthetical and move it into a note which talks about examples of how you may launch a web app, and actually link into the specs or spec drafts where possible.

The "(e.g. share data in the event of a share target launch)" parenthetical is also problematic since I don't believe we've yet defined how to put share data from a share target into a LaunchParams. It's a long-standing issue with share target that there's no programmatic way to access it (w3c/web-share-target#84) which I hope we can address with launch_handler. But for now, we can't just throw it in with an "e.g." in normative text. We can however replace it with "e.g. file data in the event of a file handling launch" since that is specified in Manifest Incubations.

At some point in the future, we should invert this and have every type of launch specified as invoking the "launch a web app" algorithm. For example, the Web Share Target spec's [Launching the web share target](https://w3c.github.io/web-share-target/#launching-the-web-share-target) algorithm should invoke "launch a web app".

Reply:
> I think this "eg" parenthetical is too vague (i.e. not linked to appropriate specs which may or may not yet exist, and is non-exhaustive) to be appropriate in normative text. I would remove the parenthetical and move it into a note which talks about examples of how you may launch a web app, and actually link into the specs or spec drafts where possible.

Done.

> 
> The "(e.g. share data in the event of a share target launch)" parenthetical is also problematic since I don't believe we've yet defined how to put share data from a share target into a LaunchParams. It's a long-standing issue with share target that there's no programmatic way to access it ([w3c/web-share-target#84](https://github.com/w3c/web-share-target/issues/84)) which I hope we can address with launch_handler. But for now, we can't just throw it in with an "e.g." in normative text. We can however replace it with "e.g. file data in the event of a file handling launch" since that is specified in Manifest Incubations.

The data may not yet be available but a basic LaunchParams with the targetURL would still be enqueued.

> At some point in the future, we should invert this and have every type of launch specified as invoking the "launch a web app" algorithm. For example, the Web Share Target spec's [Launching the web share target](https://w3c.github.io/web-share-target/#launching-the-web-share-target) algorithm should invoke "launch a web app".

:+1: